### ISSUE TITLE
OpenCL performance difference between arm_benchmark and examples using same model 

### ISSUE THREAD

**Issue by ysh329** [ARM-software/ComputeLibrary#311]
I ran alexnet using `examples/graph_alexnet` with optional `Target` 1 (0 = NEON, 1 = OpenCL) and benchmark cases (`CL/SYSTEM_TEST/AlexNet@DataType=F32:Batches=1`) as below:
```shell
yuanshuai@firefly:~/code/ComputerLibrary-17.12/build/tests$ ./arm_compute_benchmark --list-tests | grep SYSTEM
[0, ALL, ACTIVE] CL/SYSTEM_TEST/AlexNet@DataType=F16:Batches=1
[1, ALL, ACTIVE] CL/SYSTEM_TEST/AlexNet@DataType=F16:Batches=4
[2, ALL, ACTIVE] CL/SYSTEM_TEST/AlexNet@DataType=F16:Batches=8
[3, ALL, ACTIVE] CL/SYSTEM_TEST/AlexNet@DataType=F32:Batches=1
[4, ALL, ACTIVE] CL/SYSTEM_TEST/AlexNet@DataType=F32:Batches=4
[5, ALL, ACTIVE] CL/SYSTEM_TEST/AlexNet@DataType=F32:Batches=8
[6, ALL, ACTIVE] CL/SYSTEM_TEST/LeNet5@Batches=1
[7, ALL, ACTIVE] CL/SYSTEM_TEST/LeNet5@Batches=4
[8, ALL, ACTIVE] CL/SYSTEM_TEST/LeNet5@Batches=8
[9, ALL, ACTIVE] CL/SYSTEM_TEST/MobileNet@Batches=1
[10, ALL, ACTIVE] CL/SYSTEM_TEST/MobileNet@Batches=4
[11, ALL, ACTIVE] CL/SYSTEM_TEST/MobileNet@Batches=8
[12, ALL, ACTIVE] CL/SYSTEM_TEST/MobileNetV1_224@Batches=1
```

The `examples/graph_alexnet` result of inference time is `0.247441`, however the result of `arm_compute_benchmark` for `alexnet@FP32:bsize=1` is `0.136934`. So Strange!!!!!!

## examples/graph_alexnet

```shell
yuanshuai@firefly:~/code/ComputerLibrary-17.12/build/examples$ taskset -c 4 ./graph_alexnet 1

./graph_alexnet

Usage: ./graph_alexnet 1 [path_to_data] [image] [labels]

No data folder provided: using random values

Time used: 0.247441

Test passed
```

## arm_compute_benchmark
```shell
AlexNet@DataType=F16:Batches=1  0.114563
AlexNet@DataType=F32:Batches=1  0.136934
```
Besides, mobilenet also has the performance loss between `examples` and `arm_compute_benchmark`.

**Comment by AnthonyARM** [comment]
Do you observe similar execution times if you call "run" several times ?
The first run is usually slower because we need to reshape the weights

**Comment by ysh329** [comment]
@AnthonyARM Thanks my big brother, I forgot it. I'll try later! 😀

**Comment by ysh329** [comment]
@AnthonyARM I tried with a problem: I ran each example 11 times. However, the 1st, 4th, 7th of 10 times are so fast, strange! 

Below is my code added in `./example/graph_*.c`:
```cc
    double s_time = timer();
    graph.run();
    double e_time = timer();
    std::cout << "Time used: " << e_time - s_time << std::endl;

    for (int ridx = 0; ridx < 10; ridx++) {
        s_time = timer();
        graph.run();
        e_time = timer();
        std::cout << ridx << "Time used: " << e_time - s_time << std::endl;
    }
```

The execution results are below:
```shell
$ ./graph_mobilenet 1                                                                                                                                                                            [22/90]

./graph_mobilenet

Usage: ./graph_mobilenet 1 [path_to_data] [image] [labels]

No data folder provided: using random values

Time used: 0.220446
0Time used: 0.183237
1Time used: 0.000885963
2Time used: 0.176579
3Time used: 0.177057
4Time used: 0.000503063
5Time used: 0.177596
6Time used: 0.177103
7Time used: 0.0008111
8Time used: 0.176596
9Time used: 0.177247

Test passed
yuanshuai@firefly:~/code/ComputerLibrary-17.12/build/examples$ ./graph_squeezenet 1

./graph_squeezenet

Usage: ./graph_squeezenet 1 [path_to_data] [image] [labels]

No data folder provided: using random values

Time used: 0.179796
0Time used: 0.167973
1Time used: 0.000462055
2Time used: 0.166664
3Time used: 0.165501
4Time used: 0.000305891
5Time used: 0.177851
6Time used: 0.170221
7Time used: 0.0012331
8Time used: 0.166191
9Time used: 0.167294

Test passed
yuanshuai@firefly:~/code/ComputerLibrary-17.12/build/examples$ ./graph_alexnet 1

./graph_alexnet

Usage: ./graph_alexnet 1 [path_to_data] [image] [labels]

No data folder provided: using random values

Time used: 0.256728
0Time used: 0.140293
1Time used: 0.00079608
2Time used: 0.140693
3Time used: 0.137981
4Time used: 0.000552893
5Time used: 0.141899
6Time used: 0.139774
7Time used: 0.00067997
8Time used: 0.140914
9Time used: 0.140361

Test passed
yuanshuai@firefly:~/code/ComputerLibrary-17.12/build/examples$ ./graph_vgg16 1

./graph_vgg16

Usage: ./graph_vgg16 1 [path_to_data] [image] [labels]

No data folder provided: using random values

Time used: 3.90138
0Time used: 3.63174
1Time used: 0.00145102
2Time used: 3.65831
3Time used: 3.67454
4Time used: 0.00602293
5Time used: 3.66178
6Time used: 3.61321
7Time used: 0.00813103
8Time used: 3.65153
9Time used: 3.68021

Test passed
```

**Comment by AnthonyARM** [comment]
Same issue as [here](https://github.com/ARM-software/ComputeLibrary/issues/313#issuecomment-353603000)

**Comment by AnthonyARM** [comment]
Fixed in 18.01: closing.
